torch/csrc/jit/codegen/cuda/lower_warp_reduce.cpp

```diff
@@ -0,0 +1,546 @@
+#include <torch/csrc/jit/codegen/cuda/expr_evaluator.h>
+#include <torch/csrc/jit/codegen/cuda/kernel_ir_builder.h>
+#include <torch/csrc/jit/codegen/cuda/lower2device.h>
+#include <torch/csrc/jit/codegen/cuda/lower_utils.h>
+#include <torch/csrc/jit/codegen/cuda/lower_warp_reduce.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+namespace {
+
+//! A simple DCE for eliminating the
+//!  parallel broadcasts that has been fused
+//!  and their corresponding allocates
+class eliminateDeadBroadcastAndAllocate {
```

Comment:
This looks good, but is it difficult to do generic DCE rather than just do that for broadcast output tensors?

Reply:
Sounds fair.